index.js

```diff
@@ -289,6 +291,10 @@ function normalizeArguments(url, opts) {
 		opts
 	);
 
+	if (!opts.decompress) {
+		opts.encoding = null;
+	}
```

Comment:
This causes a non-compressed response to be a Buffer, even a plain text response. Is that really wanted/expected?